### ISSUE TITLE
Empty iterable/sequence as `__slots__` isn't recognized

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2502]
**Describe the bug**
Pyright doesn't give an error for assigning attributes to instances of a class with an empty iterable or sequence as its `__slots__`.

**Expected behavior**
Pyright should give an error when trying to assign attributes to instances of a class with empty slots.

**Screenshots or Code**
```python
class A:
    __slots__ = ()  # same with [] and other empty sequences/iterables

    def __init__(self):
        self.foo = 1  # 'A' object has no attribute 'foo'; no pyright error

a = A()

class B:
    __slots__ = ('foo',)  # or 'foo'

    def __init__(self):
        self.bar = 1  # "bar" is not specified in __slots__ (correct)

b = B()
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line (pyright 1.1.182)
